messaging/src/test/java/org/axonframework/eventhandling/ReplayTokenTest.java

```diff
@@ -135,4 +154,1470 @@ void createReplayTokenReturnsReplayToken() {
+
+        // EXAMPLE: forgot about events that were gaps - tokenAtReset NEVER covers the newToken
+        // IS IT VALID SCENARIO? I have null token at the beginning and then advance to 9
+        @Test
+        void eventPreviouslyProcessedShouldBeReplayEvenWhenGapsFilledDuringReplay() {
+            // tokenAtReset at index 10 with gaps at 7,8
+            // Before reset, processor saw: 0,1,2,3,4,5,6,9,10 (NOT 7,8)
+            TrackingToken tokenAtReset = GapAwareTrackingToken.newInstance(10, setOf(7L, 8L));
+            TrackingToken currentToken = ReplayToken.createReplayToken(tokenAtReset, null);
+
+            // Now event 9 arrives - currentToken has NO gaps (they were filled)
+            // Event 9 WAS processed before reset, so it SHOULD be a replay
+            TrackingToken newToken = GapAwareTrackingToken.newInstance(9, emptySet());
+            TrackingToken result = ((ReplayToken) currentToken).advancedTo(newToken);
+
+            assertInstanceOf(ReplayToken.class, result, "Should still be in replay mode");
+            assertTrue(ReplayToken.isReplay(result),
+                    "Event 9 was processed before reset, should be marked as replay even though gaps were filled");
+        }
+
+        @Test
+        void eventPreviouslyProcessedShouldBeReplayEvenWhenGapsFilledDuringReplayStartNotNull() {
+            TrackingToken tokenAtReset = GapAwareTrackingToken.newInstance(10, setOf(7L, 8L));
+            TrackingToken currentToken = ReplayToken.createReplayToken(tokenAtReset, new GapAwareTrackingToken(0L, emptyList()));
+
+            TrackingToken newToken = GapAwareTrackingToken.newInstance(9, emptySet());
+            TrackingToken result = ((ReplayToken) currentToken).advancedTo(newToken);
+
+            assertInstanceOf(ReplayToken.class, result, "Should still be in replay mode");
+            assertTrue(ReplayToken.isReplay(result),
+                    "Event 9 was processed before reset, should be marked as replay even though gaps were filled");
+        }
+
+        @Test
+        void eventAtResetIndexShouldBeReplayEvenWhenGapsFilledDuringReplay() {
+            // tokenAtReset at index 10 with gaps at 7,8
+            TrackingToken tokenAtReset = GapAwareTrackingToken.newInstance(10, setOf(7L, 8L));
+            TrackingToken currentToken = ReplayToken.createReplayToken(tokenAtReset, null);
+
+            // During replay, gaps get filled, process 0-9
+            // Event 10 - at reset index, WAS processed before reset
+            TrackingToken newToken = GapAwareTrackingToken.newInstance(10, emptySet());
+            TrackingToken result = ((ReplayToken) currentToken).advancedTo(newToken);
+
+            assertInstanceOf(ReplayToken.class, result, "Should still be in replay mode at reset index");
+            assertTrue(ReplayToken.isReplay(result),
+                    "Event 10 was processed before reset, should be marked as replay");
+        }
+
+        @Test
+        void replayBeforeResetIndexAdnGaps() {
```

Comment:
A new test, but entirely different test naming scheme from the once above...why? Make the cognitive load of validating if everything is correct here just that bit higher.